### ISSUE TITLE
How to add multiple detection heads (for small & large objects) in YOLOv8 for medical imaging?

### ISSUE THREAD

**Issue by DoniaDriss12** [ultralytics/ultralytics#20019]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi, I'm an engineering student currently working on an object detection project in the medical field, and I'm using YOLOv8 as the base architecture.

🧠 What I'm Trying to Do
I would like to add two custom detection heads to the YOLOv8 model:

-One head focused on very small objects.

-One head focused on medium to large objects.

This would allow the model to better detect across very different object scales, which is essential in medical applications.

🛠️ What I Need Help With
Could you please provide a detailed guide or high-level steps to how to do this ? 

-Which source code files should be modified (and their paths)?
-How to add an additional detection head to the model?
-Do I need to modify any other parts of the code like ?
-Can I Fine-tune an existing YOLOv8 model after adding the new head? Or do I need to train from scratch?
 etc etc ..

🧪 My Use Case
Domain: Medical imaging
Dataset: Custom dataset with bounding boxes, containing both small and medium/large medical features.

🙏 Thank You
I really appreciate any help or pointers you can give.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @DoniaDriss12, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, as well as details on model customization. Many of the most common questions may already be answered there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us assist you better.

For your custom training ❓ Question, adding multiple detection heads is an advanced feature. Please provide more details about your scenario, such as code snippets, error logs (if any), and more specific requirements. This will help us better understand your use case and guide you. In the meantime, ensure you've reviewed our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) for optimizing custom training.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or, dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package, including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to ensure you're working with the latest features and bug fixes:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response, but don't worry—an Ultralytics engineer will assist you further soon. Let us know how we can help! 😊

**Comment by Y-T-G** [comment]
There are already 3 different FPN scales for detecting small, medium and large. If you want to detect even smaller objects, you can use P2 models.

https://www.reddit.com/r/Ultralytics/comments/1gbyc47/detecting_objects_that_are_extra_small_or_extra/

**Comment by Lishumuzixin** [comment]
> There are already 3 different FPN scales for detecting small, medium and large. If you want to detect even smaller objects, you can use P2 models.
> 
> https://www.reddit.com/r/Ultralytics/comments/1gbyc47/detecting_objects_that_are_extra_small_or_extra/
Can yolov8n-p2.yaml  be used for segmentation?

**Comment by Y-T-G** [comment]
Yes, if you change the head to Segment head
